dev-tools/reconfigurator-cli/src/main.rs

```diff
@@ -124,7 +124,7 @@ impl ReconfiguratorSim {
         builder.set_external_dns_version(parent_blueprint.external_dns_version);
 
         for (_, zone) in parent_blueprint
-            .all_omicron_zones(BlueprintZoneFilter::ShouldBeRunning)
+            .all_omicron_zones(BlueprintZoneDisposition::is_in_service)
```

Comment:
With the fact that rust lets you pass named functions to closures like this, this change feels like an obvious win. The maintenance overhead is now lower for `BlueprintZoneDisposition`. All the information you need for filtering zones is in the enum (or it's impl) rather than a separate type.